CMakeLists.txt

```diff
@@ -4,6 +4,8 @@ if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
 endif ()
 
+project (rippled)
```

Comment:
moving `project` here breaks the hacky way of overriding the compiler using the `target` argument. Since I've removed that target stuff in https://github.com/ripple/rippled/pull/3249, I think this should be fine as long as both changes get merged around the same time.